src/App.tsx

```diff
@@ -1,11 +1,87 @@
 import './installSesLockdown';
-import 'ses';
+import { motion } from 'framer-motion';
+import { useEffect } from 'react';
+import { useAtom } from 'jotai';
+import { E, ERef } from '@endo/eventual-send';
+import { makeAsyncIterableFromNotifier as iterateNotifier } from '@agoric/notifier';
+
+import {
+  brandToInfoAtom,
+  walletAtom,
+  pursesAtom,
+  offersAtom,
+} from 'store/store';
+import WalletConnection from 'components/WalletConnection';
+import { INTER_LOGO } from 'assets/assets';
+import Swap from 'components/Swap';
+
+const watchPurses = async (
+  wallet: ERef<any>,
+  setPurses: (purses: any) => void,
+  mergeBrandToInfo: (entries: Iterable<Iterable<any>>) => void
+) => {
+  const n = await E(wallet).getPursesNotifier();
+  for await (const purses of iterateNotifier(n)) {
+    setPurses(purses);
+    purses.forEach(
+      ({
+        brand,
+        displayInfo,
+        brandPetname: petname,
+      }: {
+        brand: any;
+        displayInfo: any;
+        brandPetname: string;
+      }) => {
```

Comment:
I think this would read more clearly as a for loop.

Reply:
Done